source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -398,6 +399,53 @@ def tag_ecologic(resp):
                 j.tags.append('ecologic')
 
 
+def tag_special_event(instance, pb_resp):
+    for j in pb_resp.journeys:
+        # Tag only Walking solution proposed by using additional_parameters in the matrix od_additional_parameters
+        if (
+            len(j.sections) == 1
+            and j.sections[0].type == response_pb2.STREET_NETWORK
+            and j.sections[0].street_network.mode == response_pb2.Walking
+        ):
+            origin = j.sections[0].origin.uri
+            des = j.sections[0].destination.uri
+            if instance.get_od_additional_parameters(origin, des):
+                j.tags.append(SPECIAL_EVENT)
+            continue
+
+        any_section_from_to_in_od = False
+        all_lines_in_od = True
+        do_tag = False
+        for s in j.sections:
+            # Solution with more than one sections:
+            # 1. line.uri as well as destination of at least one PT exist in the matrix od_allowed_ids
+            # OR
+            # 2. line.uri of all as well as origin or destination of one PT should exist in the matrix od_allowed_ids
+            if s.type == response_pb2.PUBLIC_TRANSPORT:
+                origin = s.origin.stop_point.stop_area.uri
+                des = s.destination.stop_point.stop_area.uri
+                line_uri = s.uris.line
+
+                # Test if destination sa_uri exists in the set instance.od_stop_areas
+                destination_in_od = instance.uri_in_od_stop_areas(des)
+                # Test if line_uri is present in the set instance.od_lines
+                line_in_od = instance.uri_in_od_lines(line_uri)
+                # Tag if line.uri as well as destination sa_uri exist in corresponding lists, do not continue
+                if line_in_od and destination_in_od:
+                    do_tag = True
+                    break
```

Comment:
I may be wrong, but I think this mean you will tag the journeys A-B-C or A-B in https://navitia.atlassian.net/wiki/spaces/NAV/pages/12162302066/Itin+raires+sp+cifiques+en+cas+d+obligations+pr+fectorales althought the doc says they should not be tagged ?

Reply:
:+1: